src/QsCompiler/Tests.Compiler/TestCases/LambdaLifting.qs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft Corporation.
```

Comment:
Some ideas for additional test cases:

- Lambda that uses its arguments, e.g. `x -> x` or `(x, y) -> (y, x)`
- Lambda that uses both its arguments and enclosed variables, e.g. `let a = 0; let f = x -> (a, x);`
- Lambda that discards parameters, e.g. `(x, _, _) -> x`.

Reply:
I can make tests for the first two suggestions. But for the third bullet point, lambda's can't be defined with missing parameters.

Reply:
As per [this comment](https://github.com/microsoft/qsharp-compiler/pull/1302#discussion_r791036804) I will look into adding tests that cover all three of these bullet points.

Reply:
added